php/util/class-fieldmanager-util-gutenberg.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Class file for Fieldmanager_Util_Gutenberg
+ *
+ * @package Fieldmanager
+ */
+
+/**
+ * Gutenberg helpers.
+ */
+class Fieldmanager_Util_Gutenberg {
+	public function __construct() {
+		add_filter( 'fm_enqueue_scripts', [ $this, 'add_gutenberg_js_deps' ], 99 );
+
+		// Gutenberg sidebar polyfill.
+		fm_add_script( 'fieldmanager-gutenberg-polyfill', 'js/fieldmanager-gutenberg-support.js' );
+	}
+
+	/**
+	 * Shim for fields loaded inside of Gutenberg editor.
+	 *
+	 * Adds JS dependency to all fm scripts to ensure proper load order.
+	 *
+	 * @param array $scripts
+	 * @return array
+	 */
+	public function add_gutenberg_js_deps( $scripts ) {
+		$is_gutenberg_editor = false;
+
+		 // Do we have access to current screen?
+		if ( did_action( 'current_screen' ) ) {
+			$current_screen = get_current_screen();
+			$is_gutenberg_editor = $current_screen instanceof WP_Screen ? $current_screen->is_block_editor : false;
+		}
+
+		// Fallback if we don't have access to `current_screen`.
+		if ( ! $is_gutenberg_editor ) {
```

Comment:
This block will still run even when the screen is available if `$current_screen->is_block_editor` is `false`.

Reply:
Even if `$current_screen->is_block_editor` is false, we'd hit the fallback which returns false also in my testing, which prevented anything from loading. 🤔 This seems like what it's supposed to do, no?

Reply:
Sure, I just mean that running the fallback seems redundant if `WP_Screen` already provided the answer.

Reply:
Oh, right. I agree. Tbh I'm not certain why we have the fallback specifically since all posts should have access to it, but @GFargo might have more information for that.